### PR TITLE
Type mapper changes

### PR DESCRIPTION
**Pull request by keizer619**
Struct modify notification  support for type mapper
jsTree references imported from requirejs
typemapper switching bug fixes

### CODE REVIEW

File: modules/web/js/ballerina/views/type-mapper-definition-view.js [hunk 1]
```diff
@@ -459,6 +467,17 @@ define(['lodash', 'log', './ballerina-view', './variables-view', './type-struct-
                 }
             });
 
+            registerListeners(self, self.getSourceInfo().sourceStruct);
+
+            _.each(self.getSourceInfo().sourceStruct.getChildren(), function (variableDef) {
+                _.each(predefinedStructs, function (predefinedStruct) {
+                    if (variableDef.getTypeName() === predefinedStruct.getStructName()) {
```
Comment (hemikak) [inline modules/web/js/ballerina/views/type-mapper-definition-view.js]: Better if we can use _.isEqual() method.

File: modules/web/js/ballerina/views/type-mapper-definition-view.js [hunk 2]
```diff
@@ -494,6 +513,16 @@ define(['lodash', 'log', './ballerina-view', './variables-view', './type-struct-
                     return false;
                 }
             });
+            registerListeners(self, self.getTargetInfo().targetStruct);
+
+            _.each(self.getTargetInfo().targetStruct.getChildren(), function (variableDef) {
+                _.each(predefinedStructs, function (predefinedStruct) {
+                    if (variableDef.getTypeName() === predefinedStruct.getStructName()) {
```
Comment (hemikak) [inline modules/web/js/ballerina/views/type-mapper-definition-view.js]: Better if we can use _.isEqual() method.

File: modules/web/js/ballerina/views/type-mapper-definition-view.js [hunk 3]
```diff
@@ -505,6 +534,61 @@ define(['lodash', 'log', './ballerina-view', './variables-view', './type-struct-
         };
 
         /**
+         * register listeners to be fired upon changes in structs
+         * @param view
+         * @param struct
+         */
```
Comment (hemikak) [inline modules/web/js/ballerina/views/type-mapper-definition-view.js]: Shall we fill in the documentation ?
http://usejsdoc.org/tags-param.html

File: modules/web/package.json [hunk 4]
```diff
@@ -52,7 +52,6 @@
     "jquery-ui": "^1.12.1",
     "js-yaml": "^3.7.0",
     "jsplumb": "^2.2.8",
-    "jstree": "^3.3.2",
```
Comment (hemikak) [inline modules/web/package.json]: Any reason for removing form package.json ?

Comment (keizer619) [inline modules/web/package.json]: @hemikak ; importing jstree and its style.css has broken the stylings of the type mapper, so this has been reverted back to previous state to load from requirejs until this is properly being fixed.

**Pull request merged**